relnotes/reverse.feature.md

```diff
@@ -0,0 +1,5 @@
+* `ocean.core.array.Iteration`
+
+  New module containing single function for now - `reverse`, which allows to
+  iterate given slice in reverse order without any allocations (similar to
+  built-in `foreach_reverse`).
```

Comment:
Most of its downsides is that it doesn't to any type control, happily resorting to plain `foreach` semantics for anything that isn't an array. That makes it rather error-prone in scope of refactorings especially.